modules/widgets/migrate-to-core/image-widget.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * Migration from Jetpack's Image Widget to WordPress' Core Image Widget.
+ *
+ * @since 4.9
+ *
+ * @package Jetpack
+ */
+
+/**
+ * Migrates all active instances of Jetpack's image widget to Core's media image widget.
+ */
+function jetpack_migrate_image_widget() {
+	// Only trigger the migration from wp-admin
+	if ( ! is_admin() ) {
+		return;
+	}
+	// Only migrate if the new widget is available and we haven't yet migrated
+	if ( ! class_exists( 'WP_Widget_Media_Image' ) || Jetpack_Options::get_option( 'image_widget_migration' ) ) {
+		return;
+	}
+
+	$default_data = array(
+		'attachment_id' => 0,
+		'url' => '',
+		'title' => '',
+		'size' => 'full',
+		'width' => 0,
+		'height' => 0,
+		'align' => 'none',
+		'caption' => '',
+		'alt' => '',
+		'link_type' => '',
+		'link_url' => '',
+		'image_classes' => '',
+		'link_classes' => '',
+		'link_rel' => '',
+		'image_title' => '',
+		'link_target_blank' => false,
+	);
+
+	$media_image      = get_option( 'widget_media_image' );
+	$sidebars_widgets = wp_get_sidebars_widgets();
+
+	foreach ( get_option( 'widget_image' ) as $id => $widget ) {
+		if ( is_string( $id ) ) {
+			continue;
+		}
+
+		$media_image[ $id ] = array_merge( $default_data, array_intersect_key( $widget, $default_data ), array(
+			'alt'         => $widget['alt_text'],
+			'height'      => $widget['img_height'],
+			'image_title' => $widget['img_title'],
```

Comment:
To port over the alignment, we could do something like this: 

`image_classes => 'align' . $widget['align'],`